packages/node/src/indexer/api.service.ts

```diff
@@ -134,53 +131,40 @@ export class ApiService implements OnApplicationShutdown {
       });
     }
     this.patchApiAt(this.patchedApi);
-    this.patchApiQuery(this.patchedApi);
+    if (blockHash) {
+      await this.patchApiQuery(this.patchedApi, blockHash);
+    }
     this.patchApiTx(this.patchedApi);
     this.patchApiQueryMulti(this.patchedApi);
     this.patchDerive(this.patchedApi);
     this.patchApiRpc(this.patchedApi);
     (this.patchedApi as any).isPatched = true;
   }
 
-  async setBlockhash(blockHash: BlockHash, inject = false): Promise<void> {
+  async setBlockhash(blockHash: BlockHash): Promise<void> {
     if (!this.patchedApi) {
       await this.getPatchedApi();
     }
+    const { registry } = await this.api.getBlockRegistry(blockHash);
     this.currentBlockHash = blockHash;
-    if (inject) {
-      const { metadata, registry } = await this.api.getBlockRegistry(blockHash);
-      this.patchedApi.injectMetadata(metadata, true, registry);
-      this.patchApi(registry);
-    }
-  }
-
-  private replaceToAtVersion(
-    original: QueryableStorageEntry<'promise' | 'rxjs', AnyTuple>,
-    atMethod: string,
-  ) {
-    return (...args: any[]) => {
-      return original[atMethod](this.currentBlockHash, ...args);
-    };
+    await this.patchApi(registry, blockHash);
   }
 
   private redecorateStorageEntryFunction(
     original: QueryableStorageEntry<'promise' | 'rxjs', AnyTuple>,
     apiType: 'promise' | 'rxjs',
   ): QueryableStorageEntry<'promise' | 'rxjs', AnyTuple> {
-    const newEntryFunc = this.replaceToAtVersion(
-      original,
-      'at',
-    ) as QueryableStorageEntry<'promise' | 'rxjs', AnyTuple>;
+    const newEntryFunc = original;
     newEntryFunc.at = NOT_SUPPORT('at');
     newEntryFunc.creator = original.creator;
-    newEntryFunc.entries = this.replaceToAtVersion(original, 'entriesAt');
+    newEntryFunc.entries = original.entries;
```

Comment:
the var name `original` is confusing, it used to be `original` but now more like `decorated`

Reply:
Maybe these lines can be removed now, as `const newEntryFunc = original;`